invokeai/app/api/routers/models.py

```diff
@@ -2,35 +2,59 @@
 
 
 import pathlib
-from typing import List, Literal, Optional, Union
+from enum import Enum
+from typing import Any, List, Literal, Optional, Union
 
 from fastapi import Body, Path, Query, Response
 from fastapi.routing import APIRouter
 from pydantic import BaseModel, parse_obj_as
 from starlette.exceptions import HTTPException
 
+from invokeai.app.api.dependencies import ApiDependencies
 from invokeai.backend import BaseModelType, ModelType
-from invokeai.backend.model_management import MergeInterpolationMethod
-from invokeai.backend.model_management.models import (
+from invokeai.backend.model_manager import (
     OPENAPI_MODEL_CONFIGS,
+    DuplicateModelException,
     InvalidModelException,
-    ModelNotFoundException,
+    ModelConfigBase,
     SchedulerPredictionType,
+    UnknownModelException,
 )
-
-from ..dependencies import ApiDependencies
+from invokeai.backend.model_manager.download import DownloadJobStatus, UnknownJobIDException
+from invokeai.backend.model_manager.merge import MergeInterpolationMethod
 
 models_router = APIRouter(prefix="/v1/models", tags=["models"])
 
-UpdateModelResponse = Union[tuple(OPENAPI_MODEL_CONFIGS)]
-ImportModelResponse = Union[tuple(OPENAPI_MODEL_CONFIGS)]
-ConvertModelResponse = Union[tuple(OPENAPI_MODEL_CONFIGS)]
-MergeModelResponse = Union[tuple(OPENAPI_MODEL_CONFIGS)]
-ImportModelAttributes = Union[tuple(OPENAPI_MODEL_CONFIGS)]
+# NOTE: The generic configuration classes defined in invokeai.backend.model_manager.config
+# such as "MainCheckpointConfig" are repackaged by code originally written by Stalker
+# into base-specific classes such as `abc.StableDiffusion1ModelCheckpointConfig`
+# This is the reason for the calls to dict() followed by pydantic.parse_obj_as()
+
+# There are still numerous mypy errors here because it does not seem to like this
+# way of dynamically generating the typing hints below.
+InvokeAIModelConfig: Any = Union[tuple(OPENAPI_MODEL_CONFIGS)]
```

Comment:
Mypy is happy with the `: Any` for me - there are other issues reported in this file, but this isn't one of them.

Pylance, however, does not like this. Not sure how to quiet it without littering `# type: ignore` everywhere.

Reply:
Not seeing any mypy errors. Just orange warnings about docstrings. Any other solution for pylance? There must be a lot of complaints.

Reply:
There are a decent number of complaints (mypy and pylance catch different things). In this case, there doesn't appear to be any way to silence pylance.

Reply:
I don't understand why this is needed in the first place. The model config classes are already pydantic models, why can't we just put them in a union directly and let the schema generation use that?

I *think* that prior to this PR, the models were generated by this logic, but now they are all explicitly defined, so this rigamarole is superfluous.